### PR TITLE
Master mass mailing responsive mail dep

### PR DESCRIPTION
**Pull request by dep-odoo**
Task: https://www.odoo.com/web#id=1836556&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
Pad: https://pad.odoo.com/p/r.ba1da71959d9618cc815fe3d81c1a07b

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by Yajo** [comment]
It's great to see that #10954 is getting attention finally!

I just wanted to point out (just in case you missed that part) that given [the current transcoder](https://github.com/odoo/odoo/blob/11.0/addons/web_editor/static/src/js/editor/transcoder.js) is in JS, some styles can get lost, like any style starting with `-moz-` while you use chrome, or vice versa.

This part should be done in the backend too, to avoid all those problems. Possibly using https://github.com/peterbe/premailer or something similar.

File: addons/mass_mailing/models/mass_mailing.py [hunk 1]
```diff
@@ -819,3 +820,12 @@ def _process_mass_mailing_queue(self):
                 mass_mailing.send_mail()
             else:
                 mass_mailing.state = 'done'
+
+    def append_style(self, body):
+        files, remains = self.env['ir.qweb']._get_asset_content('mass_mailing.assets_mail_themes_common', {})
+        assets = AssetsBundle('mass_mailing.assets_mail_themes_common', files, remains, env=self.env)
+
+        scss_css_styles = assets.preprocess_css()
+        style_in_head = "<head><style>" + scss_css_styles + "</style></head>"
```
Comment (chienandalu) [inline addons/mass_mailing/models/mass_mailing.py]: It'd be great to compose this in a separate method to be able to insert compatibility snippets (for outlook sake mainly) in the header like:
```html
<!--[if gte mso 15]>
        <xml>
            <o:OfficeDocumentSettings>
            <o:AllowPNG/>
            <o:PixelsPerInch>96</o:PixelsPerInch>
            </o:OfficeDocumentSettings>
        </xml>
        <![endif]-->
```

**Comment by pga-odoo** [comment]
LGTM

**Comment by Gorash** [comment]
Good :+1: 
We need a test for who use this part:
https://github.com/odoo/odoo/pull/24572/files#diff-68c153bbbb29470138a51e3b8cf503c6R79
And check the content of the email that was sent

File: addons/mass_mailing/wizard/mail_compose_message.py [hunk 2]
```diff
@@ -57,6 +57,7 @@ def get_mail_values(self, res_ids):
 
             blacklist = self._context.get('mass_mailing_blacklist')
             seen_list = self._context.get('mass_mailing_seen_list')
+            mass_mail_layout = self.env.ref('mass_mailing.mass_mailing_mail_layout')
```
Comment (Gorash) [inline addons/mass_mailing/wizard/mail_compose_message.py]: Need to add:
raise_if_not_found= False

(In order not to have any error and to send the mail (to take the same direction as the commit: https://github.com/odoo/odoo/commit/f8031a61f3885d2279e6490258e181dbb8d3130c )

Comment (dep-odoo) [reply addons/mass_mailing/wizard/mail_compose_message.py]: @Gorash  :+1:  done

**Comment by Gorash** [comment]
Merged: https://github.com/odoo/odoo/commit/967d228a44080381d185c252ff29c993b2b88429

**Pull request closed without merging**